server/src/main/java/com/broadcom/lsp/cobol/service/CobolTextDocumentService.java

```diff
@@ -197,8 +198,15 @@ public void didChange(DidChangeTextDocumentParams params) {
   public void didClose(DidCloseTextDocumentParams params) {
     String uri = params.getTextDocument().getUri();
     LOG.info(format("Document closing invoked on URI %s", uri));
+    interruptAnalysis(uri);
+    communications.cancelProgressNotification(uri);
```

Comment:
This might give an indication to user that the processing is stopped, while its not. Do we want to handle this in method interceptor?

Reply:
Yes, it should be part of aspect

Reply:
I should not be a part of the aspect, it will lead only to the spreading of logic. Please, encapsulate all the Communications usages in the service.

Reply:
Yes, it does make the code less clean. Adding it in aspect has brought some extra complexity and spread the responsibility.
Should have discussed before implementing :(